src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/util/JwtProvider.kt

```diff
@@ -0,0 +1,41 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.util
+
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.SignatureAlgorithm
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.core.io.ByteArrayResource
+import org.springframework.stereotype.Component
+import java.security.KeyStore
+import java.security.PrivateKey
+import java.util.Base64
+
+@Component
+class JwtProvider(
```

Comment:
Maybe take a look at https://github.com/ministryofjustice/hmpps-nomis-prisoner-api/blob/main/src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/JwtAuthHelper.kt as an alternative approach which means you wouldn't need `hmpps-auth-dev-key` stuff when running tests

Reply:
Interesting, I'm guessing this works because this is overriding the part of Spring Security that would normally use the JWKS from HMPPS Auth with one generated at runtime?

```
@Bean
  fun jwtDecoder(): JwtDecoder = NimbusJwtDecoder.withPublicKey(keyPair.public as RSAPublicKey).build()
```

Reply:
Have switched to this method 👍